streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala

```diff
@@ -103,4 +109,37 @@ object KubernetesNativeApplicationClient extends KubernetesNativeClientTrait {
     flinkConf.safeSet(DeploymentOptions.TARGET, ExecutionMode.KUBERNETES_NATIVE_APPLICATION.getName)
     super.doTriggerSavepoint(request, flinkConf)
   }
+
+  def convertFlinkDeploymentDef(
+      submitRequest: SubmitRequest,
+      flinkConfig: Configuration): FlinkDeploymentDef = {
+    val spec = FlinkDeploymentDef(
+      name = submitRequest.appName,
+      namespace = submitRequest.k8sSubmitParam.kubernetesNamespace,
+      image = "flink:" + submitRequest.flinkVersion.majorVersion,
+      flinkVersion = converFlinkVersion(submitRequest.flinkVersion.majorVersion),
+      jobManager =
+        JobManagerDef(cpu = 1, memory = flinkConfig.toMap.get("taskmanager.memory.process.size")),
+      taskManager =
+        TaskManagerDef(cpu = 1, memory = flinkConfig.toMap.get("jobmanager.memory.process.size")),
+      job = Some(
+        JobDef(
+          jarURI =
+            submitRequest.buildResult.asInstanceOf[DockerImageBuildResponse].dockerInnerMainJarPath,
+            parallelism = submitRequest.properties.get("parallelism.default").asInstanceOf[Int]
+        ))
+    )
+    spec
+  }
+
+  def converFlinkVersion(version: String): FlinkVersion = {
+    version match {
+      case "1.17" => FlinkVersion.V1_17
+      case "1.16" => FlinkVersion.V1_16
+      case "1.15" => FlinkVersion.V1_15
+      case "1.14" => FlinkVersion.V1_14
+      case "1.13" => FlinkVersion.V1_13
+      case _ => null
```

Comment:
Do we need to throw an exception?

Reply:
Can scala's option type be used to send exceptions